agent/api/task.go

```diff
@@ -210,7 +214,7 @@ func (task *Task) initializeCredentialsEndpoint(credentialsManager credentials.M
 func (task *Task) addNetworkResourceProvisioningDependency() {
 	// TODO check networking mode for the task before doing this
 	for _, container := range task.Containers {
-		if container.Name == emptyHostVolumeName && container.IsInternal {
+		if container.IsInternal {
```

Comment:
Neither. I'm working on making `IsInternal` an enum. It would be more clear when I update the PR with that commit. The short answer is that the only Internal container we have today is the empty host volume container.